### ISSUE TITLE
Actions not completing, throwing error

### ISSUE THREAD

**Issue by NetTecture** [object/Simple.OData.Client#384]
I hav a problem with simlple actions not completing.

This is a snippet from the metadata:

<EnumType Name="OpsServiceOperation">
<Member Name="Start" Value="0"/>
<Member Name="Stop" Value="1"/>
</EnumType>
</Schema>
<Schema xmlns="http://docs.oasis-open.org/odata/ns/edm" Namespace="Default">
<Function Name="Snapshot" IsBound="true">
<Parameter Name="bindingParameter" Type="Reflexo.Api.TrdConnection"/>
<Parameter Name="Symbol" Type="Edm.String" Unicode="false"/>
<ReturnType Type="Reflexo.Api.TrdInstrumentSnapshot"/>
</Function>
<Action Name="Operate" IsBound="true">
<Parameter Name="bindingParameter" Type="Reflexo.Api.TrdInstance"/>
<Parameter Name="Operation" Type="Reflexo.Api.OpsServiceOperation" Nullable="false"/>
</Action>
<Action Name="Operate" IsBound="true">
<Parameter Name="bindingParameter" Type="Reflexo.Api.TrdProvider"/>
<Parameter Name="Operation" Type="Reflexo.Api.OpsServiceOperation" Nullable="false"/>
</Action>

**Comment by object** [comment]
Yes it looks like a failure handling the action response but there are too few details listed here for me to understand what's going on. I would like to have an OData server metadata document and a copy of the actual JSON response that causes this error.

**Comment by NetTecture** [comment]
I will see what I can do.

**Comment by damnlogin** [comment]
You should not return `Ok()` if you are not going to return any content IMHO - use `return NoContent()`  that will produce HTTP response with code 202 and no content will be expected by the consuming party, so the the issue should be gone.

**Comment by thePmr** [comment]
Hm. What if you don't own the backend and cannot change 200 to 204? Moreover RFC states that 200 may come with zero-length payload.

**Comment by damnlogin** [comment]
> Hm. What if you don't own the backend and cannot change 200 to 204? Moreover RFC states that 200 may come with zero-length payload.

Sorry but what RFC you are referring to? If I am not mistaken rfc1945 states body can be empty in response to HEAD request only.

**Comment by thePmr** [comment]
This one: https://www.rfc-editor.org/rfc/rfc7231#section-6.3.1

> Aside from responses to CONNECT, a 200 response always has a payload, though an origin server MAY generate a payload body of zero length.
